### ISSUE TITLE
Request support for array of class with array

### ISSUE THREAD

**Issue by victoryang00** [alibaba/yalantinglibs#278]
### What happened + What you expected to happen
I want to support something like
```c++
template<size_t out>
struct stat{
    std::array<int, out> sb;
    auto operator==(const stat& rhs) const {
        for (int i = 0; i < out; i++) {
            if (sb[i] != rhs.sb[i]) {
                return false;
            }
        }
        return true;
    }
    stat(){
        for (int i = 0; i < out; i++) {
            sb[i] = rand() % 100;
        }
    }
};

template <size_t in,size_t out>
struct person {
  int age;
  std::array<stat<out>, in>state ;
  std::string name;

  auto operator==(const person& rhs) const {
    for (int i = 0; i < in; i++) {
      printf("%d\n", state[i]);
      if (state[i] != rhs.state[i]) {
        return false;
      }
    }
    return age == rhs.age && name == rhs.name;
  }
};
```
It will be filtered out by [line](https://github.com/alibaba/yalantinglibs/blob/7857ba02e880dd4f9f83869c3ab49da7ded3fd92/include/struct_pack/struct_pack/struct_pack_impl.hpp#L559), and if I remove the line, the following checks will comply later for `std::is_aggregate_v`. But according to the fixed_size compile time definition, the array in the array class will be sized during compile time, you just need to make extra checks for if visit the member of the class with an array, if it's aggregate for every member in the struct, and make extra logic during generating visit_member function. 
### Reproduction way
just compile the above code in [code](https://github.com/victoryang00/yalantinglibs/blob/main/src/struct_pack/examples/serialize/serialize.cpp)

### Anything else
If you think it's doable, we can make it together.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by poor-circle** [comment]
For now, the user-defined class should be aggregate. You need remove the constructor of struct `stat`.
```cpp
template <size_t out>
struct stat1 {
  std::array<int, out> sb = init_sb();
  friend bool operator==(const stat1&, const stat1&) = default;
  static auto init_sb() {
    std::array<int, out> sb;
    for (int i = 0; i < out; i++) {
      sb[i] = rand() % 100;
    }
    return sb;
  }
};

template <size_t in, size_t out>
struct person1 {
  int age;
  std::array<stat1<out>, in> state;
  std::string name;
  friend bool operator==(const person1&, const person1&) = default;
};
```

**Comment by victoryang00** [comment]
My stupid, thx

**Comment by poor-circle** [comment]
The non-aggregate user-defined class will be supported later. But you need add some macro for reflection.

**Comment by poor-circle** [comment]
Fixed by #279. Now we support non-aggregate type.
But you need add some reflection info manually :
```cpp
template <size_t out>
constexpr std::size_t struct_pack::members_count<stat1<out>> = 1;

template <size_t in, size_t out>
constexpr std::size_t struct_pack::members_count<person1<in, out>> = 3;
```

**Comment by yanminhui** [comment]
> Fixed by #279. Now we support non-aggregate type. But you need add some reflection info manually :
> 
> ```c++
> template <size_t out>
> constexpr std::size_t struct_pack::members_count<stat1<out>> = 1;
> 
> template <size_t in, size_t out>
> constexpr std::size_t struct_pack::members_count<person1<in, out>> = 3;
> ```

```.cpp
#include <iostream>

namespace struct_pack {
template <typename T>
constexpr size_t members_count = 2023;
} // namespace struct_pack

namespace ns {
struct Foo {};

// ERROR: PLACE IN NS !!!
template <>
constexpr size_t struct_pack::members_count<Foo> = 23;

} // namespace ns

int main() {
    std::cout << struct_pack::members_count<ns::Foo> << std::endl;
}
```

not a good idea, because it couldn't place `members_count` in user namespace, and it will intrude into `struct_pack`.
